src/audio/google/google_rtc_audio_processing.c

```diff
@@ -80,28 +106,130 @@ void GoogleRtcFree(void *ptr)
 	return rfree(ptr);
 }
 
-static void google_rtc_audio_processing_params(struct processing_module *mod,
-					       struct sof_source *ref,
-					       struct sof_source *mic,
-					       struct sof_sink *out)
+static ALWAYS_INLINE float s16_to_float(const char *ptr)
```

Comment:
Why are we not using the macros [here](https://github.com/thesofproject/sof/blob/b3b9abffe7ee85bf0884d392d3d243161efadbbe/src/include/sof/audio/format.h#L71)?

Reply:
(Again, please continue review in the main PR, but answering here):

This was discussed earlier: Those generate horrifyingly pessimal code (runtime division!) and can't be fixed without breaking the ABI (because as specified they work for 64 bit values).  These seem like they're intended for generating compile time constants for fixed point use.  There's also a pcm_convert API elsewhere in SOF, but that's designed to work in a block on linear arrays of samples (and don't expose inlinable single-sample conversions) where here we need to unify de/interleaving with the conversion.

Basically: SOF just doesn't have an API for this.  I could move this somewhere else and try to unify with an existing library.